server/rest/src/main/java/org/infinispan/rest/resources/CounterResource.java

```diff
@@ -0,0 +1,140 @@
+            .invocation().methods(GET).path("/v2/counters/{counterName}").handleWith(this::getCounter)
+            .invocation().methods(DELETE).path("/v2/counters/{counterName}").handleWith(this::resetCounter)
+            .create();
+   }
+
+   private NettyRestResponse getCounter(RestRequest request) throws RestResponseException {
+      try {
+         String counterName = request.variables().get("counterName");
+         String accept = request.getAcceptHeader();
+         MediaType contentType = accept == null ? MediaType.TEXT_PLAIN : negotiateMediaType(accept);
+         NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+         CounterConfiguration configuration = counterManager.getConfiguration(counterName);
+         if (configuration == null) {
+            return responseBuilder.status(HttpResponseStatus.NOT_FOUND.code()).entity("Counter not found").build();
+         } else {
+            responseBuilder.contentType(contentType);
+            switch (configuration.type()) {
+               case WEAK:
+                  long value = counterManager.getWeakCounter(counterName).getValue();
+                  responseBuilder.entity(Long.toString(value));
+                  break;
+               case BOUNDED_STRONG:
+               case UNBOUNDED_STRONG:
+                  value = counterManager.getStrongCounter(counterName).getValue().get(1, TimeUnit.MINUTES);
+                  responseBuilder.entity(Long.toString(value));
+                  break;
+            }
+         }
+         return responseBuilder.header(HttpHeaderNames.CACHE_CONTROL.toString(), CacheControl.noCache()).build();
+      } catch (Exception e) {
+         throw new RestResponseException(e);
+      }
+   }
+
+   private MediaType negotiateMediaType(String accept) {
+      return MediaType.parseList(accept).filter(t ->
+            t.match(MediaType.TEXT_PLAIN) // TODO: add more types
+      ).findFirst().orElseThrow(() -> Log.REST.unsupportedDataFormat(accept));
+   }
+
+
+   private RestResponse addCounter(RestRequest request) {
+      try {
+         String counterName = request.variables().get("counterName");
+         NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+         CounterConfiguration configuration = counterManager.getConfiguration(counterName);
+         if (configuration == null) {
+            return responseBuilder.status(HttpResponseStatus.NOT_FOUND.code()).entity("Counter not found").build();
+         } else {
+            ContentSource contents = request.contents();
+            long delta;
+            if (contents == null) {
+               delta = 1;
+            } else {
+               String s = contents.asString();
+               delta = Long.parseLong(s);
+            }
+            switch (configuration.type()) {
+               case WEAK:
+                  counterManager.getWeakCounter(counterName).add(delta);
```

Comment:
I think it should wait for the `add()` to complete. At least to make sure there was no exception...